catalog/spatial/registry/registry-rest-endpoint/src/main/java/org/codice/ddf/registry/rest/endpoint/RegistryRestEndpoint.java

```diff
@@ -0,0 +1,171 @@
+
+    private RegistryPublicationService registryPublicationService;
+
+    private RegistryReportHelper reportHelper;
+
+    public RegistryRestEndpoint() {
+        templateLoader = new ClassPathTemplateLoader();
+        templateLoader.setPrefix("/templates");
+        templateLoader.setSuffix(".hbt");
+    }
+
+    @Path("/{registryId}/publication/{sourceId}")
+    @POST
+    public Response publish(@PathParam("registryId") String registryId,
+            @PathParam("sourceId") String destinationId) {
+        if (StringUtils.isBlank(registryId) || StringUtils.isBlank(destinationId)) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity("Registry and destination ID cannot be blank")
+                    .build();
+        }
+
+        try {
+            registryPublicationService.publish(registryId, destinationId);
+        } catch (FederationAdminException e) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity("Error publishing node to destination " + destinationId)
+                    .build();
+        }
+
+        return Response.status(Response.Status.OK)
+                .build();
+    }
+
+    @Path("/{registryId}/publication/{sourceId}")
+    @DELETE
+    public Response unpublish(@PathParam("registryId") String registryId,
+            @PathParam("sourceId") String destinationId) {
+        if (StringUtils.isBlank(registryId) || StringUtils.isBlank(destinationId)) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity("Registry and destination ID cannot be blank")
+                    .build();
+        }
+
+        try {
+            registryPublicationService.unpublish(registryId, destinationId);
+        } catch (FederationAdminException e) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity("Error publishing node to destination " + destinationId)
+                    .build();
+        }
+
+        return Response.status(Response.Status.OK)
+                .build();
+    }
+
+    @Path("/{registryId}/report")
+    @Produces(MediaType.TEXT_HTML)
+    @GET
+    public Response viewRegInfoHtml(@PathParam("registryId") final String registryId,
+            @QueryParam("sourceId") final List<String> sourceIds) {
+        String html = "";
+        if (StringUtils.isBlank(registryId)) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity("Registry Id cannot be blank")
+                    .build();
+        }
+
+        RegistryPackageType registryPackage;
+        try {
+            registryPackage = federationAdminService.getRegistryObjectByRegistryId(registryId,
+                    sourceIds);
+        } catch (FederationAdminException e) {
+            String message = "Error getting registry package.";
+            LOGGER.error("{} For metacard id: '{}', optional sources: {}",
```

Comment:
Should this log say `metacard` id or `registry` id? Seems like you are actually logging the `registry` id.